test/test_autograd.py

```diff
@@ -7003,6 +7013,234 @@ def test_size_check(self):
 
             dual = fwAD.make_dual(foo, tangent[1:])
 
+    # The following test functions want to ensure all the following behaviors:
+    #   - Ensure that default level system in the python binding works
+    #   - Ensure that only level 0 exists and nesting is properly disabled
+    #   - Ensure that printing works fine
+    #   - Ensure that basic packing/unpacking works
+    #   - Ensure that advanced packing/unpacking works
+    #     - For memory / version counter share
+    #     - For backward AD (regular ops)
+    #   - Ensure that view + inplace for both modes work fine
+    #   - Ensure we do proper cleanup on exit of a level
+
+    def test_default_level(self):
+        foo = torch.rand(2)
+        bar = torch.rand(2)
+
+        with fwAD.dual_level():
+            baz = fwAD.make_dual(foo, bar)
+            baz_primal, baz_tangent = fwAD.unpack_dual(baz)
+        self.assertEqual(baz_primal, foo)
+        self.assertIs(baz_tangent, bar)
```

Comment:
This seems like an oddly precise thing to assert. You really want to commit to tangent having an is-a relationship after this change?

Reply:
I don't expect this to be a hard constraint. It is more that I added the most restrictive test possible for the current behavior to be sure to know if/when we change it.
Do you prefer me to relax this and just assert equality?

Reply:
At the very least, you'd say "this doesn't really matter, feel free to relax" in comments here. But ideally, you'd spec what this is. Since `baz_primal` doesn't preserve identity it seems weird for tangent to do so, but maybe it's fine. Hard to say.